libservo is busted, but CI doesn’t catch it

in [linux.yml](https://github.com/servo/servo/blob/26e724bd83792ccb647f7b7afb4d86ddf14d38c3/.github/workflows/linux.yml) and other platforms we have:

```
cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked --all-targets
```

--all-targets builds both the lib target and the test targets, so both [dependencies] and [dev-dependencies] of libservo take effect. but people consuming libservo as a library (or using analysis tools like [cargo-public-api](https://github.com/cargo-public-api/cargo-public-api)) are building the lib target only, so only [dependencies] take effect.

turns out, that more realistic build configuration has been busted all along:

```
$ cargo build -p servo --locked
   Compiling servo-net v0.0.1 (/cuffs/code/servo/components/net)
error[E0432]: unresolved import `rustls::crypto::aws_lc_rs`
  --> components/net/connector.rs:28:38
   |
28 | use rustls::crypto::{CryptoProvider, aws_lc_rs};
   |                                      ^^^^^^^^^ no `aws_lc_rs` in `crypto`
   |
note: found an item that was configured out
  --> /home/delan/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rustls-0.23.37/src/crypto/mod.rs:30:9
   |
29 | #[cfg(feature = "aws_lc_rs")]
   |       --------------------- the item is gated behind the `aws_lc_rs` feature
30 | pub mod aws_lc_rs;
   |         ^^^^^^^^^
```

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -288,7 +288,7 @@ jobs:
           # We can remove this after upgrading to Rust 1.90, where lld will be the default.
           RUSTFLAGS: "-C link-arg=-fuse-ld=lld"
         run: |
-          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked --all-targets
+          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked
 
   # Runs the underlying job (“workload”) on a self-hosted runner if available.
   runner-select-coverage:

--- a/.github/workflows/mac-arm64.yml
+++ b/.github/workflows/mac-arm64.yml
@@ -249,4 +249,4 @@ jobs:
           toolchain: ${{ steps.msrv.outputs.rust_version }}
       - name: Compile libservo with MSRV
         run: |
-          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked --all-targets
+          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked

--- a/.github/workflows/mac.yml
+++ b/.github/workflows/mac.yml
@@ -257,4 +257,4 @@ jobs:
           toolchain: ${{ steps.msrv.outputs.rust_version }}
       - name: Compile libservo with MSRV
         run: |
-          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked --all-targets
+          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -266,4 +266,4 @@ jobs:
           toolchain: ${{ steps.msrv.outputs.rust_version }}
       - name: Compile libservo with MSRV
         run: |
-          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked --all-targets
+          cargo +${{ steps.msrv.outputs.rust_version }} build -p servo --locked

--- a/components/net/Cargo.toml
+++ b/components/net/Cargo.toml
@@ -61,7 +61,7 @@ quick_cache = { version = "0.6.18", default-features = false, features = ["ahash
 regex = { workspace = true }
 resvg = { workspace = true }
 rustc-hash = { workspace = true }
-rustls = { workspace = true }
+rustls = { workspace = true, features = ["aws-lc-rs"] }
 rustls-pki-types = { workspace = true }
 rustls-platform-verifier = { workspace = true }
 serde = { workspace = true }

--- a/components/servo/Cargo.toml
+++ b/components/servo/Cargo.toml
@@ -151,7 +151,7 @@ http-body-util = { workspace = true }
 hyper = { workspace = true }
 servo = { path = ".", features = ["tracing"] }
 net = { package = "servo-net", path = "../net", features = ["test-util"] }
-rustls = { version = "0.23", default-features = false, features = ["aws-lc-rs"] }
+rustls = { version = "0.23", default-features = false }
 tracing = { workspace = true }
 winit = { workspace = true }
 

